src/plugins/plot/chart/MctChart.vue

```diff
@@ -200,7 +200,13 @@ export default {
     this.chartVisible = true;
     this.chartContainer = this.$refs.chart;
     this.drawnOnce = false;
-    this.visibilityObserver = new IntersectionObserver(this.visibilityChanged);
+    const rootContainer = document.querySelector('.js-main-container');
```

Comment:
I think this is ok to not be a $ref? $refs lookup is much quicker and it's reactive, but we wouldn't be running this repeatedly, and I don't think it needs to be reactive?

either way we should leave a comment here referencing the issue number and explaining why we're using querySelector